File: Array/1894-merge-strings-alternately/merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder s = new StringBuilder();

        int n1 = word1.length();
        int n2 = word2.length();
        int i=0;

        while (i<n1 || i<n2){
            if(i<n1) s.append(word1.charAt(i));
            if(i<n2) s.append(word2.charAt(i));
            i++;
        }

        return s.toString();
    }
}

File: Array/1894-merge-strings-alternately/merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        merged = ''

        l1, l2 = 0, 0
        n1, n2 = len(word1), len(word2)

        for idx, c in enumerate(word1):
            merged += c
            if idx < len(word2):
                merged += word2[idx]
        
        for i in range(idx+1, len(word2)):
            merged += word2[i]

        return merged

